cmake/template/YARPTargetsStatic.cmake.in

```diff
@@ -0,0 +1,66 @@
+# Copyright (C) 2006-2018 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+#
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+
+# For static builds, all the dependencies used by targets in YARPTargets.cmake
+# must be available before including the file, but due to the recursive
+# dependencies in YARP_init, we are forced to use a hack
+
+# Pretend to be looking for a different package (YARP_static_hack)
+set(_CMAKE_FIND_PACKAGE_NAME ${CMAKE_FIND_PACKAGE_NAME})
+set(CMAKE_FIND_PACKAGE_NAME YARP_static_hack)
+
+# Include the targets file. This will set the YARP_static_hack_NOT_FOUND_MESSAGE
+# variable because some targets are still missing.
+include("${CMAKE_CURRENT_LIST_DIR}/YARPTargets.cmake")
+
+# Include all targets files for the missing targets found in the
+# YARP_static_hack_NOT_FOUND_MESSAGE variable
+# Accumulate all targets not found in the _expected_targets variable
+string(REGEX REPLACE ".+:  (.+)" "\\1" _targets "${YARP_static_hack_NOT_FOUND_MESSAGE}")
```

Comment:
Not a big deal since it's just a hack, but CMake defines a `YARP_static_hack_NOT_FOUND_MESSAGE_targets` variable with just the list of targets. Perhaps it could make things (a bit) simpler.